### PR TITLE
Fixed aircraft reserving influence when they are unable to land

### LINKED ISSUE #20370: Crash related idle behavior of dead aircraft
Mod: SP

```
Exception of type `System.InvalidOperationException`: Attempted to get trait from destroyed object (stormrider 1564 (not in world))
   at OpenRA.TraitDictionary.CheckDestroyed(Actor actor) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\TraitDictionary.cs:line 84
   at OpenRA.Mods.Common.Traits.Locomotor.UpdateCellBlocking(CPos cell) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Traits\World\Locomotor.cs:line 488
   at OpenRA.Mods.Common.Traits.Locomotor.CanMoveFreelyInto(Actor actor, CPos cell, SubCell subCell, BlockedByActor check, Actor ignoreActor) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Traits\World\Locomotor.cs:line 230
   at OpenRA.Mods.Common.Pathfinder.DensePathGraph.GetConnections(CPos position) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Pathfinder\DensePathGraph.cs:line 131
   at OpenRA.Mods.Common.Pathfinder.PathSearch.Expand() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Pathfinder\PathSearch.cs:line 219
   at OpenRA.Mods.Common.Pathfinder.PathSearch.FindPath() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Pathfinder\PathSearch.cs:line 290
   at OpenRA.Mods.Common.Pathfinder.HierarchicalPathFinder.FindPath(Actor self, IReadOnlyCollection`1 sources, CPos target, BlockedByActor check, Int32 heuristicWeightPercentage, Func`2 customCost, Actor ignoreActor, Boolean laneBias, PathFinderOverlay pathFinderOverlay) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Pathfinder\HierarchicalPathFinder.cs:line 755
   at OpenRA.Mods.Common.Traits.PathFinder.FindPathToTargetCell(Actor self, IEnumerable`1 sources, CPos target, BlockedByActor check, Func`2 customCost, Actor ignoreActor, Boolean laneBias) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Traits\World\PathFinder.cs:line 89
   at OpenRA.Mods.Common.Activities.MoveAdjacentTo.CalculatePathToTarget(Actor self, BlockedByActor check) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Activities\Move\MoveAdjacentTo.cs:line 131
   at OpenRA.Mods.Common.Activities.Move.EvalPath(BlockedByActor check) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Activities\Move\Move.cs:line 114
   at OpenRA.Mods.Common.Activities.Move.OnFirstRun(Actor self) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Activities\Move\Move.cs:line 129
   at OpenRA.Activities.Activity.TickOuter(Actor self) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Activities\Activity.cs:line 111
   at OpenRA.Traits.ActivityUtils.RunActivity(Actor self, Activity act) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Traits\ActivityUtils.cs:line 31
   at OpenRA.Activities.Activity.TickChild(Actor self) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Activities\Activity.cs:line 147
   at OpenRA.Mods.Common.Activities.MoveAdjacentTo.Tick(Actor self) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Mods.Common\Activities\Move\MoveAdjacentTo.cs:line 103
   at OpenRA.Activities.Activity.TickOuter(Actor self) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Activities\Activity.cs:line 111
   at OpenRA.Traits.ActivityUtils.RunActivity(Actor self, Activity act) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Traits\ActivityUtils.cs:line 31
   at OpenRA.Activities.Activity.TickChild(Actor self) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Activities\Activity.cs:line 147
   at OpenRA.Activities.Activity.TickOuter(Actor self) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Activities\Activity.cs:line 111
   at OpenRA.Traits.ActivityUtils.RunActivity(Actor self, Activity act) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Traits\ActivityUtils.cs:line 31
   at OpenRA.Actor.Tick() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Actor.cs:line 262
   at OpenRA.World.Tick() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\World.cs:line 441
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 636
   at OpenRA.Game.LogicTick() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 651
   at OpenRA.Game.Loop() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 820
   at OpenRA.Game.Run() in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 873
   at OpenRA.Game.InitializeAndRun(String[] args) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Game\Game.cs:line 308
   at OpenRA.Launcher.Program.Main(String[] args) in D:\work\Projects\Github\shattered-paradise-sdk-mirror\engine\OpenRA.Launcher\Program.cs:line 32
```

I apply the [engine](https://github.com/MustaphaTR/OpenRA/commit/02d9ba020dc91939160568d38bceec444d61d859) with https://github.com/MustaphaTR/OpenRA/pull/61 but it should not be the cause of the bug right?

### PR DESCRIPTION
**Pull request by Mailaender**
Resolves https://github.com/OpenRA/OpenRA/issues/20370

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs [hunk 1]
```diff
@@ -191,6 +191,8 @@ IEnumerable<ActorInit> IActorPreviewInitInfo.ActorPreviewInits(ActorInfo ai, Act
 
 		bool IOccupySpaceInfo.SharesCell => false;
 
+		public bool CanLand => LandableTerrainTypes.Count > 0;
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: This seems wrong. In Aircraft trait there's already a CanLand function that we use to determine wether we can land

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs [hunk 2]
```diff
@@ -880,7 +885,8 @@ public void RemoveInfluence()
 
 		public bool HasInfluence()
 		{
-			return landingCells.Any() || self.World.Map.DistanceAboveTerrain(CenterPosition).Length < Info.MinAirborneAltitude;
+			var altitude = self.World.Map.DistanceAboveTerrain(CenterPosition).Length;
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: why waste a variable? It should be just within the return

Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: Just to don't overflow the line. I think the compiler will optimize something like this.

**Comment by RoosterDragon** [comment]
Was this superseded by #20401?

This check seems supect to me, as I would expect the `aircraft.CanLand` call which checks against the landable terrain types to have been called first.

Or maybe if `!CanLand(...)` then we should throw as calling code probably shouldn't try and add influence if the aircraft can't land there.

**Pull request closed without merging**